app/code/Magento/LoginAsCustomerGraphQl/Model/Resolver/RequestCustomerToken.php

```diff
@@ -0,0 +1,116 @@
+use Magento\Framework\GraphQl\Exception\GraphQlNoSuchEntityException;
+use Magento\Framework\GraphQl\Query\Resolver\ContextInterface;
+use Magento\Framework\GraphQl\Query\Resolver\Value;
+use Magento\Framework\GraphQl\Query\ResolverInterface;
+use Magento\Framework\GraphQl\Schema\Type\ResolveInfo;
+use Magento\LoginAsCustomerApi\Api\ConfigInterface;
+use Magento\LoginAsCustomerGraphQl\Model\LoginAsCustomer\CreateCustomerToken;
+
+/**
+ * Gets customer token
+ *
+ * Class RequestCustomerToken
+ */
+class RequestCustomerToken implements ResolverInterface
+{
+    /**
+     * @var ConfigInterface
+     */
+    private $config;
+
+    /**
+     * @var AuthorizationInterface
+     */
+    private $authorization;
+
+    /**
+     * @var CreateCustomerToken
+     */
+    private $createCustomerToken;
+
+    /**
+     * RequestCustomerToken constructor.
+     * @param AuthorizationInterface $authorization
+     * @param ConfigInterface $config
+     * @param CreateCustomerToken $createCustomerToken
+     */
+    public function __construct(
+        AuthorizationInterface $authorization,
+        ConfigInterface $config,
+        CreateCustomerToken $createCustomerToken
+    ) {
+        $this->authorization = $authorization;
+        $this->config = $config;
+        $this->createCustomerToken = $createCustomerToken;
+    }
+
+    /**
+     * Get Customer Token using email
+     *
+     * @param Field $field
+     * @param ContextInterface $context
+     * @param ResolveInfo $info
+     * @param array|null $value
+     * @param array|null $args
+     * @return Value|mixed|void
+     * @throws GraphQlAuthorizationException|GraphQlNoSuchEntityException|LocalizedException
+     */
+    public function resolve(
+        Field $field,
+        $context,
+        ResolveInfo $info,
+        array $value = null,
+        array $args = null
+    ) {
+        $isAllowedLogin = $this->authorization->isAllowed('Magento_LoginAsCustomer::login');
+        $isEnabled = $this->config->isEnabled();
+
+        /* Get input params */
+        try {
+            $args = $args['input'] ?: [];
+        } catch (NoSuchEntityException $e) {
+            throw new GraphQlNoSuchEntityException(__('Check input params.'));
+        }
+
+        if (empty(trim($args['customer_email'], " "))) {
+            throw new GraphQlInputException(__('Specify the "customer email" value.'));
+        }
+
+        $this->validateUser($context);
+
+        if (!$isAllowedLogin || !$isEnabled) {
+            throw new GraphQlAuthorizationException(
+                __('Login as Customer is disabled.')
+            );
+        }
+
+        return $this->createCustomerToken->execute($args['customer_email']);
+    }
+
+    /**
+     * Check if its an admin user
+     *
+     * @param $context
+     * @throws GraphQlAuthorizationException
+     */
+    private function validateUser($context)
+    {
+        if ($context->getUserType() != 2 || $context->getUserId() == 0) {
```

Comment:
Please, use strict comparison
```suggestion
        if ($context->getUserType() !== 2 || $context->getUserId() === 0) {
```